### ISSUE TITLE
Incomplete/parameterized where clause generated by .where()

### ISSUE THREAD

**Issue by maxwondercorn** [tgriesser/knex#2361]
# Node 8.9/Express

Knex version: 0.14.1
Database + version: mysql/mariaDb using mySQL2 driver 1.4.2
OS: osx - DB is running on remote server

# Bug

I have a SELECT query that Knex correctly generates but when I try to add a WHERE clause, the generated SQL has a variable assignment.

So this query:
```
   SELECT 
            unit.id, 
             SUBSTR(uid, 4) AS uid, 
             unit.type_name, 
             product.name AS product, 
             designator, 
             color_name, 
             status.name AS status_name, 
             metric, 
             rule.uom_name as uom, 
             reset, 
             rule_name, 
             last_scan,
             scanned, 
             extra
  FROM 
            unit 
            JOIN product ON (unit.productID = product.id)
            LEFT JOIN status ON (unit.status = status.id)
            JOIN rule ON (rule_name = rule.name)
  ORDER BY
             `uid` ASC
```
is generated as follow:

```
   const cols = `unit.id, SUBSTR(uid, 4) AS uid, unit.type_name, product.name AS product, designator, 
                    color_name, status.name AS status_name, metric, rule.uom_name as uom, reset, rule_name, 
                    last_scan, scanned, extra`;`

   knex(table).select(knex.raw(cols))
      .join('product', 'unit.productID', '=', 'product.id')
      .leftJoin('status', 'unit.status', '=', 'status.id')
      .join('rule', 'rule_name', '=', 'rule.name')
      .orderBy('uid', 'ASC')
      .then((records) => res.send(records))
      .catch((error) => res.sendStatus(404));
```

The correct SQL query is generated and data retrieved.  If I try to add a .where() the resulting query has an unassigned variable in the WHERE clause.

```
  knex(table).select(knex.raw(cols))
     .join('product', 'unit.productID', '=', 'product.id')
     .leftJoin('status', 'unit.status', '=', 'status.id')
     .join('rule', 'rule_name', '=', 'rule.name')
     .where({sitecode: "ALB"})
     .orderBy('uid', 'ASC')
     .then((records) => res.send(records))
     .catch((error) => res.sendStatus(404));
```

which generates:

```
select 
	unit.id, 
	SUBSTR(uid, 4) AS uid, 
	unit.type_name, 
	product.name AS product, 
	designator, 
	color_name, 
	status.name AS status_name, 
	metric, 
	rule.uom_name as uom, 
	reset, 
	rule_name, 
	last_scan, 
	scanned, 
	extra 
from 
	`unit` 
	inner join `product` on `unit`.`productID` = `product`.`id` 
	left join `status` on `unit`.`status` = `status`.`id` 
	inner join `rule` on `rule_name` = `rule`.`name` 
where 
	`sitcode` = ? 
order by 
	`uid` ASC
```

I've tried the variants .where('site code', 'ALB'), .where('site code', '=', 'ALB'), and whereRaw('sitcode = ?', ['ALB'])  all of which generate the same result.

Other than the 404 for no values returned, there are no error messages generated by Knex.

**Comment by elhigu** [comment]
All that you pasted looks correct. How are you running the queries? https://runkit.com/embed/dcgbmjizxsay

Am I missing something, I don't see any bug here.

**Comment by maxwondercorn** [comment]
Okay, so my first mistake was not understanding that .where() generates a statement with bindings.  It wasn't obvious from the docs and I hadn't looked at any of my working Selects using .where().  I had some pretty ugly queries that Knex has handled without issue.

Being a bit smarter, I looked at the back end and realized that data inconsistency was causing the failure.  The query was implemented differently than documented.  Implementation had LIKE instead of equals in the were statements.  Thanks
